modules/navier_stokes/include/actions/AddNavierStokesVariablesAction.h

```diff
@@ -0,0 +1,43 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef ADDNAVIERSTOKESVARIABLESACTION_H
+#define ADDNAVIERSTOKESVARIABLESACTION_H
+
+#include "NSAction.h"
+
+class AddNavierStokesVariablesAction;
+
+template<>
+InputParameters validParams<AddNavierStokesVariablesAction>();
+
+/**
+ * This class allows us to have a section of the input file like the
+ * following which automatically adds all the required nonlinear
+ * variables with the appropriate scaling.
+ *
+ * [NavierStokes]
```

Comment:
The team is thinking about organizing all module syntax under a main block: https://github.com/idaholab/moose/issues/5585